src/persist-txn/src/txn_cache.rs

```diff
@@ -223,7 +223,11 @@ impl<T: Timestamp + Lattice + TotalOrder + StepForward + Codec64> TxnsCacheState
             // Not registered, maybe it will be in the future? In the meantime,
             // treat it like a normal shard (i.e. pass through reads) and check
             // again later.
-            return ReadDataTo(self.progress_exclusive.clone());
+            if ts < self.progress_exclusive {
+                return ReadDataTo(self.progress_exclusive.clone());
```

Comment:
Isn't this unreachable due to the `assert` above on line 208?

> `assert!(self.progress_exclusive >= ts);`

Reply:
The args are swapped between the two, so I _think_ this is fine, but definitely confusing! I'll think about how to make this more readable and push something up once Aljoscha's comments come in

Reply:
Ah ok, no, it's probably fine to leave it, or swap the arguments. I think it's just towards the end of the day here.

Reply:
Ended up leaving this. All the ts comparisons in this method are currently `ts <op> <frontier>` but all the "precondition" asserts are `<frontier> <op> ts`. I'm a little torn switching to either of them fully because the latter is more how partial orders work (and they line up with the `update_{gt,ge}` methods that satisfy the preconditions), but the former is probably more readable.